dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/java2sec/PermissionManager.java

```diff
@@ -339,7 +339,11 @@ private CodeSource createCodeSource(String codeBase) {
         Certificate[] certs = null;
         CodeSource codeSource = null;
         try {
-            codeSource = new CodeSource(new URL("wsjar:file:/" + codeBase), certs);
+            if (wsjarUrlStreamHandlerAvailable) {
+                codeSource = new CodeSource(new URL("wsjar:file:/" + codeBase), certs);
+            } else {
+                codeSource = new CodeSource(new URL("jar:file:/" + codeBase), certs);
```

Comment:
fallback should happen in the catch block for `MalformedURLException` I think without checking for the .`wsjarUrlStreamHandlerAvailable` flag.